contracts/layer-2/NFTBurner.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: MIT
+// Unagi Contracts v1.0.0 (NFTBurner.sol)
+pragma solidity 0.8.12;
+
+import "@openzeppelin/contracts/interfaces/IERC721.sol";
+import "@openzeppelin/contracts/interfaces/IERC721Receiver.sol";
+import "@openzeppelin/contracts/interfaces/IERC721Receiver.sol";
+import "@openzeppelin/contracts/interfaces/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+
+/**
+ * @title NFTBurner
+ * @dev TODO
+ */
+contract NFTBurner is IERC721Receiver, AccessControl {
```

Comment:
Let's not extends IERC721Receiver, it's not necessary here

Reply:
It's necessary for the burn reservation process. Specifically, for when the tokens are placed in escrow. During that process, the NFTBurner.sol acts as an ERC721Receiver.